design/spaces.go

```diff
@@ -78,6 +108,13 @@ var spaceAttributes = a.Type("SpaceAttributes", func() {
 	})
 })
 
+var spacenameValidationFunction = func() {
+	a.MaxLength(63) // maximum name length is 63 characters
+	a.MinLength(1)  // minimum name length is 1 characters
+	a.Pattern("^([A-Za-z0-9][-A-Za-z0-9]*)?[A-Za-z0-9]$")
```

Comment:
So Kubernetes follows the RFC 1123 with respect to naming the namespaces. This RFC talks about the naming of hosts, since the namespace name then becomes the route/FQDN.

So the regex for the validation seems wrong, here is a reference from upstream 
https://github.com/kubernetes/apimachinery/blob/241e268dc8e07223510442ab7d7902695c585d80/pkg/util/validation/validation.go#L106-L107

@xcoulon @kwk

Reply:
@surajssd if RFC 1123 can also handle UUIDs, then I'd says we use a space's ID in pods. @aslakknutsen is that possible? I don't see a good reason for why the pod naming conventions should block us from having more relaxed space naming conventions.